src/v1/sources/shopify/pixelTransform.js

```diff
@@ -20,6 +29,43 @@ const NO_OPERATION_SUCCESS = {
   },
   statusCode: 200,
 };
+const extractCartToken = (cartToken) => {
+  if (typeof cartToken !== 'string') {
+    logger.error(`Cart token is not a string`);
+    return undefined;
+  }
+  const cartTokenParts = cartToken.split('/');
+  return cartTokenParts[3];
+};
+
+const handleCartTokenRedisOperations = async (inputEvent, clientId) => {
+  try {
+    const cartTokenLocation = pixelEventToCartTokenLocationMapping[inputEvent.name];
+    if (!cartTokenLocation) {
+      stats.increment('shopify_pixel_cart_token_not_found', {
+        event: inputEvent.name,
+        writeKey: inputEvent.query_parameters.writeKey,
+      });
+      return;
+    }
+    const unparsedCartToken = _.get(inputEvent, cartTokenLocation);
+    const cartToken = extractCartToken(unparsedCartToken);
+    inputEvent.anonymousId = clientId;
+    if (isDefinedNotNullNotEmpty(clientId)) {
+      await RedisDB.setVal(cartToken, ['anonymousId', clientId]);
+      stats.increment('shopify_pixel_cart_token_set', {
+        event: inputEvent.name,
+        writeKey: inputEvent.query_parameters.writeKey,
+      });
+    }
+  } catch (error) {
```

Comment:
Non blocker but if we can separate the try catch for this redis set val it will be helpful

Reply:
Instead, can move the entire cartToken extraction logic from inputEvent to `extractCartToken` function

Reply:
addressed